src/js/load-pyodide.js

```diff
@@ -375,52 +362,133 @@ function restoreOrigWasmPlugin() {
  * @async
  */
 export async function loadPackage(names, messageCallback, errorCallback) {
+  messageCallback = messageCallback || console.log;
+  errorCallback = errorCallback || console.error;
   if (Module.isPyProxy(names)) {
-    let temp;
-    try {
-      temp = names.toJs();
-    } finally {
-      names.destroy();
-    }
-    names = temp;
+    names = names.toJs();
   }
-
   if (!Array.isArray(names)) {
     names = [names];
   }
-  // get shared library packages and load those first
-  // otherwise bad things happen with linking them in firefox.
-  let sharedLibraryNames = [];
-  try {
-    let sharedLibraryPackagesToLoad = recursiveDependencies(
-      names,
-      messageCallback,
-      errorCallback,
-      true
-    );
-    for (let pkg of sharedLibraryPackagesToLoad) {
-      sharedLibraryNames.push(pkg[0]);
-    }
-  } catch (e) {
-    // do nothing - let the main load throw any errors
+
+  const [toLoad, toLoadShared] = recursiveDependencies(names, errorCallback);
+  if (toLoad.size === 0 && toLoadShared.size === 0) {
+    messageCallback("No new packages to load");
+    return;
   }
 
   let releaseLock = await acquirePackageLock();
+  for (let [pkg, uri] of [...toLoad, ...toLoadShared]) {
```

Comment:
Doesn't it make sense to load the shared libraries before the other packages, though I guess it doesn't make a difference in this particular case aside from the logging order.

Reply:
I believe that the shared libraries need to be loaded first to ensure that linking works correctly (of course there is still exactly one shared library CLAPACK aside from my test library).

Reply:
But yeah, I don't think the logging order matters much?